echopype/tests/consolidate/test_consolidate_integration.py

```diff
@@ -223,6 +205,119 @@ def _create_array_list_from_echoview_mats(paths_to_echoview_mat: List[pathlib.Pa
     return list_of_mat_arrays
 
 
+@pytest.mark.parametrize(
+    ["location_type", "sonar_model", "path_model", "raw_and_xml_paths", "extras"],
+    [
+        (
+            "empty-location",
+            "EK60",
+            "EK60",
+            ("ooi/CE02SHBP-MJ01C-07-ZPLSCB101_OOI-D20191201-T000000.raw", None),
+            None,
+        ),
+        (
+            "with-track-location",
+            "EK60",
+            "EK60",
+            ("Winter2017-D20170115-T150122.raw", None),
+            None,
+        ),
+        (
+            "fixed-location",
+            "AZFP",
+            "AZFP",
+            ("17082117.01A", "17041823.XML"),
+            {'longitude': -60.0, 'latitude': 45.0, 'salinity': 27.9, 'pressure': 59},
+        ),
+    ],
+)
+def test_add_location(
+        location_type,
+        sonar_model,
+        path_model,
+        raw_and_xml_paths,
+        extras,
+        test_path
+):
+    # Prepare the Sv dataset
+    raw_path = test_path[path_model] / raw_and_xml_paths[0]
+    if raw_and_xml_paths[1]:
+        xml_path = test_path[path_model] / raw_and_xml_paths[1]
+    else:
+        xml_path = None
+
+    ed = ep.open_raw(raw_path, xml_path=xml_path, sonar_model=sonar_model)
+    env_params = None
+    if location_type == "fixed-location":
+        point_ds = xr.Dataset(
+            {
+                "latitude": (["time"], np.array([float(extras['latitude'])])),
+                "longitude": (["time"], np.array([float(extras['longitude'])])),
+            },
+            coords={
+                "time": (["time"], np.array([ed['Sonar/Beam_group1'].ping_time.values.min()]))
+            },
+        )
+        ed.update_platform(point_ds)
+
+        # AZFP require external salinity and pressure
+        if sonar_model == "AZFP":
+            env_params = {
+                'temperature': ed["Environment"]['temperature'].values.mean(),
+                'salinity': extras['salinity'],
+                'pressure': extras['pressure'],
```

Comment:
nitpicking here: could you change all your `'` to `"`?

Reply:
Will do. And yikes, I was pretty messy there, with mixed `'` and `"`!

Reply:
It's fixed.